### ISSUE TITLE
im still requires gtk 1.2

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3340]
|<img alt="doko's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| doko reported|
|-|-|
|Trac ID|trac#3340|
|Type|defect|
|Created|2008-07-08 14:12:55Z|

with 8.1.0.  gtk 1.2 isn't found anymore in recent distributions

$ im
Traceback (most recent call last):
  File "/usr/bin/im", line 21, in <module>
    im.run()
  File "/usr/lib/python2.5/site-packages/twisted/words/scripts/im.py", line 13, in run
    from twisted.internet import gtkreactor
  File "/usr/lib/python2.5/site-packages/twisted/internet/gtkreactor.py", line 23, in <module>
    pygtk.require('1.2')
  File "/var/lib/python-support/python2.5/pygtk.py", line 85, in require
    "required version '%s' not found on system" % version
AssertionError: required version '1.2' not found on system

<details><summary>Searchable metadata</summary>

```
trac-id__3340 3340
type__defect defect
reporter__doko doko
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1215526375000000 1215526375000000
changetime__1241305299000000 1241305299000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)<a name="note_1"></a>|@jyknight commented|
|-|-|

There seem to be two possible resolutions:
 * remove im
 * make im use gtk2

Also, relatedly, [#2833](https://github.com/twisted/twisted/issues/2833) (deprecate and eventually remove gtkreactor) should be done.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph set status to `closed`|
|-|-|

(In [26801]) Remove a bunch of grotty old non-working GTK1 and Swing code.

This eliminates all of the gtk1 code remaining in Twisted besides the explicit
gtk1 reactor support.

Author: glyph

Reviewer: mwh, exarkun

Fixes [#3699](https://github.com/twisted/twisted/issues/3699)

Fixes [#3340](https://github.com/twisted/twisted/issues/3340)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
